src/dune_engine/scheduler.ml

```diff
@@ -702,10 +702,10 @@ end = struct
     let fiber =
       Fiber.Var.set t_var t (fun () -> Fiber.with_error_handler f ~on_error)
     in
-    Console.Status_line.set (Fun.const None);
     match Fiber.run fiber ~iter with
     | res ->
       assert (Event.pending_jobs () = 0);
+      Console.Status_line.set (Fun.const None);
```

Comment:
That seems about right, but I feel like we should do this cleanup systematically, i.e. not just in the case of a successful termination but also in the case `exception Abort err | exception exn`

Reply:
Okay! Should I use `Fun.protect` for this, or should I just copy over the line to the other `match` branches?